### PR TITLE
testutil/promrated: add metric to count errors from rated api

### PR DESCRIPTION
**Pull request by LukeHackett12**
Rated recently added authentication and this has been failing silently as the error was just being logged

category: bug
ticket: #1738

### CODE REVIEW

File: testutil/promrated/rated.go [hunk 1]
```diff
@@ -90,6 +90,8 @@ func getValidationStatistics(ctx context.Context, ratedEndpoint string, validato
 			backoff()
 
 			continue
+		} else if res.StatusCode == http.StatusUnauthorized {
+			panic("unauthorized")
```
Comment (dB2510) [inline testutil/promrated/rated.go:94]: suggest handle this error rather than panic

Comment (LukeHackett12) [reply testutil/promrated/rated.go:94]: How would you handle it? The intention behind the panic is to start a crash loop so we can know that it isn't working. Nothing will change if we repeatedly try do this.

Comment (corverroos) [reply testutil/promrated/rated.go:94]: crashloop is not a way to do monitoring, prefer logs + metrics and then grafana alerts

Comment (LukeHackett12) [reply testutil/promrated/rated.go]: OK I added a metric for counting rated api errors if that works better 👍🏼

**Review by OisinKyne — APPROVED** [review]

**Pull request merged**